src/gui/transferlistmodel.cpp

```diff
@@ -288,8 +288,9 @@ QString TransferListModel::displayValue(const BitTorrent::Torrent *torrent, cons
         if (hideValues && (value <= 0))
             return {};
 
+         // Convert to string with rounding to match the value of the spinRatioLimit QDoubleSpinBox
          return ((static_cast<int>(value) == -1) || (value > BitTorrent::Torrent::MAX_RATIO))
-                  ? QString::fromUtf8(C_INFINITY) : Utils::String::fromDouble(value, 2);
+                  ? QString::fromUtf8(C_INFINITY) : QLocale::system().toString(value, 'f', 2);
```

Comment:
There is a reason that we prefer using our own rounding so I think this change is incorrect.
Also I cannot reproduce the mentioned bug so maybe it is already fixed!?

In case the bug persist you might need to reimplement the spinbox widget and use this https://doc.qt.io/qt-5/qdoublespinbox.html#textFromValue with `Utils::String::fromDouble()`

Reply:
The reason for the custom conversion here is to work around the case where numbers like 99.999 get rounded to 100 and files in progress are prematurely displayed as 100% complete.

Try changing the value of the spinbox by clicking and holding on the arrows key for some time to let the number spin for several seconds, then release and save the new value to reproduce it.

Why would you want to re-implement the ratio spinbox widget to change the default conversion with default rounding to make an example "3.149999999" double number value displayed as "3.14" instead of "3.15" in the spinbox text value?

Reply:
>The reason for the custom conversion here is to work around the case where numbers like 99.999 get rounded to 100 and files in progress are prematurely displayed as 100% complete.

This isn't the only use case. qbt is using it in many other places. I don't judge whether that is correct or not, it is that if you think it is wrong, then all wrong usages should be fixed at once, not just here.

>Try changing the value of the spinbox by clicking and holding on the arrows key for some time to let the number spin for several seconds, then release and save the new value to reproduce it.

OK, I can reproduce it.

In short, the QDoubleSpinBox is rounding up, while `Utils::String::fromDouble()` is rounding down.
For example:
```c++
x = 17.509000
m_ui->spinRatioLimit->setValue(x); // 17.51
Utils::String::fromDouble(x); // 17.50
```

Reply:
>In short, the QDoubleSpinBox is rounding up, while Utils::String::fromDouble() is rounding down.
For example:

@qbittorrent/bug-handlers 
So which rounding do you prefer?

Reply:
Reimplemented the spinbox widget to display its value rounded down using `Utils::String::fromDouble()`  to match the value displayed for the ratio limit in the torrents list as suggested.

There is no more discrepancy in the displayed values now, however please note that there is a new minor unexpected behaviour introduced with this kind of rounding:

After spinned for some time, the value of the spinner will end up showing a number which is not a multiple of the step value.

Ex.: with the initial value of the ratio of "1.00" and the step of "0.05" if you press and hold the arrows key for a several seconds and release, you may see the an unexpected value like "3.99", "3.14", etc. displayed whereas "4.00", "3.15" would be a more appropriate result.

Please check & LMK.

Reply:
Please check and advise why appveyor build is failing, as it builds fine for me on Linux & Windows.

Reply:
>There is no more discrepancy in the displayed values now, however please note that there is a new minor unexpected behaviour introduced with this kind of rounding:
>
>After spinned for some time, the value of the spinner will end up showing a number which is not a multiple of the step value.

This is somewhat unexpected and personally I find this behavior isn't much of an improvement of the current "value mismatch"...
I suppose we will still have to find a way to resolve it.

Reply:
Added a possibility to pass a custom round function to the number to string converter to allow rounding methods other than "round down" and changed that to "normal round" for the ratio limit display value.